chore: app router - /team, /org, /[user] booking pages (excl. embeds)

hbjORbj opened the pull request:
## What does this PR do?

- Fixes CAL-4898
- Left a self review and comments

### Tested /team sub-pages


https://github.com/user-attachments/assets/51fba0bc-61d0-45d9-88a4-4c79ddbb9724




## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Test every booking page (user, team, org)

hbjORbj committed:
update env vars

hbjORbj committed:
update middleware

hbjORbj committed:
remove pages router and move pages to /app

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into chore/app-router-team-pages

hbjORbj committed:
move to /team

hbjORbj committed:
update imports

hbjORbj committed:
fix

hbjORbj commented on apps/web/pages/org/[orgSlug]/index.ts:
```diff
@@ -1 +0,0 @@
-export { getServerSideProps, default } from "@pages/team/[slug]";
```
- renamed `.ts` to `.tsx` and copied the code from "@pages/team/[slug]" (since this is removed now)

hbjORbj commented on apps/web/pages/org/[orgSlug]/team/[slug]/index.ts:
```diff
@@ -1 +0,0 @@
-export { getServerSideProps, default } from "@pages/team/[slug]";
```
- renamed `.ts` to `.tsx` and copied the code from "@pages/team/[slug]" (since this is removed now)

hbjORbj committed:
remove pages router and move org pages to /app

hbjORbj committed:
wip

hbjORbj committed:
fix orgSlug/user pages

hbjORbj committed:
fix orgSlug/user/type pages

hbjORbj committed:
remove generateMetadata from embed pages

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into chore/app-router-team-pages

hbjORbj committed:
fix

hbjORbj committed:
remove pages router for /user pages

hbjORbj committed:
generateMetadata is not needed in embed pages

hbjORbj committed:
remove testing setups for app router migration

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into chore/app-router-team-pages

hbjORbj committed:
Merge branch 'main' into chore/app-router-team-pages

hbjORbj committed:
remove future/org

hbjORbj committed:
no layout in [user] page

hbjORbj committed:
simplify

hbjORbj committed:
fix

hbjORbj committed:
fix OG image for [user]

hbjORbj committed:
fix OG images for org/user and team/slug

hbjORbj committed:
Merge branch 'main' into chore/app-router-team-pages

hbjORbj committed:
fix OG images for booking page

hbjORbj committed:
fix all metadata

hbjORbj committed:
use isBrandingHidden

hbjORbj committed:
remove BookerSeo and its usages

hbjORbj committed:
rename excludeAppNameFromTitle -> hideBranding

hbjORbj committed:
remove logic for meeting type in HeadSeo

hbjORbj committed:
remove BookerSeo instances from team-view and users-public-view

hbjORbj committed:
create generateMeetingMetadata util and use it to reduce duplicate code

hbjORbj committed:
remove BookerSeo imports

hbjORbj committed:
fix spacing

hbjORbj committed:
remove constructMeetingImage mock from head-seo.test

hbjORbj committed:
fetch avatarUrl using user id for user page metadata

hbjORbj commented on packages/ui/components/head-seo/head-seo.test.tsx:
```diff
@@ -45,9 +45,6 @@ vi.mock("@calcom/lib/OgImages", async () => {
     constructGenericImage() {
       return "constructGenericImage";
     },
-    constructMeetingImage() {
```
- removing this because meeting prop is removed from `HeadSeo`.
- `HeadSeo` will be completely removed once we completely migrate to App Router

hbjORbj commented on packages/ui/components/head-seo/head-seo.test.tsx:
```diff
@@ -149,12 +146,4 @@ describe("Tests for HeadSeo component", () => {
     });
   });
-
-  test("Should render with meeting props", async () => {
```
- removing this because meeting prop is removed from `HeadSeo`.
- `HeadSeo` will be completely removed once we completely migrate to App Router

hbjORbj commented on apps/web/app/[user]/[type]/embed/page.tsx:
```diff
@@ -6,8 +6,6 @@ import { getServerSideProps } from "@server/lib/[user]/[type]/getServerSideProps
 
 import LegacyPage, { type PageProps } from "~/users/views/users-type-public-view";
 
-export { generateMetadata } from "../page";
```
- generateMetadata is not needed for embeds

hbjORbj commented on apps/web/app/[user]/embed/page.tsx:
```diff
@@ -7,8 +7,6 @@ import { getServerSideProps } from "@server/lib/[user]/getServerSideProps";
 import type { PageProps as UserPageProps } from "~/users/views/users-public-view";
 import LegacyPage from "~/users/views/users-public-view";
 
-export { generateMetadata } from "../page";
```
- generateMetadata is not needed for embeds

hbjORbj committed:
fix test

hbjORbj committed:
remove unused test cases

hbjORbj committed:
index and follow must be true by default

hbjORbj committed:
invert noindex/nofollow flags

hbjORbj committed:
remove HeadSeo for already migrated pages and refactor prepareMetadata

hbjORbj committed:
fix organization-settings.e2e.ts

hbjORbj committed:
fix order

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into chore/app-router-team-pages

hbjORbj committed:
enable parallel test execution for dynamic-booking e2e test

hbjORbj committed:
fix

hbjORbj committed:
+ could be %2B in app router

hbjORbj commented on packages/lib/defaultEvents.ts:
```diff
@@ -177,7 +177,7 @@ export const getUsernameList = (users: string | string[] | undefined): string[]
   users = arrayCast(users);
 
-  const allUsers = users.map((user) => user.replace(/( |%20|%2b)/g, "+").split("+")).flat();
+  const allUsers = users.map((user) => user.replace(/( |%20|%2b)/gi, "+").split("+")).flat();
```
Apparently, "+" could be "%2B" in app router

hbjORbj committed:
refactor handling logic for embeds in app router

hbjORbj committed:
fix isEmbed

hbjORbj committed:
Merge origin/main

hbjORbj committed:
fix embed-core

hbjORbj committed:
remove dead code

hbjORbj commented on apps/web/app/_utils.tsx:
```diff
@@ -103,3 +104,22 @@ export const _generateMetadata = async (
   };
 };
+
+export const generateMeetingMetadata = async (
```
The key difference is `const image = SEO_IMG_OGIMG + constructMeetingImage(meeting);`. Using constructMeetingImage is important.

hbjORbj commented on apps/web/app/team/[slug]/[type]/page.tsx:
```diff
@@ -27,12 +27,33 @@ export const generateMetadata = async ({ params, searchParams }: PageProps) => {
+  return {
+    ...metadata,
+    robots: {
+      follow: !(event?.hidden || !isSEOIndexable),
+      index: !(event?.hidden || !isSEOIndexable),
```
- `noindex` and `nofollow` are not supported in App Router. We need to use `follow` and `index` instead.

hbjORbj commented on turbo.json:
```diff
@@ -249,7 +249,6 @@
     "APP_ROUTER_AUTH_ERROR_ENABLED",
     "APP_ROUTER_AUTH_PLATFORM_ENABLED",
     "APP_ROUTER_AUTH_OAUTH2_ENABLED",
-    "APP_ROUTER_TEAM_ENABLED",
```
I confirmed that we do not have env variables for user or org

hbjORbj commented on apps/web/modules/d/[link]/d-type-view.tsx:
```diff
@@ -20,13 +19,6 @@ export default function Type({
 }: PageProps) {
   return (
     <main className={getBookerWrapperClasses({ isEmbed: !!isEmbed })}>
-      <BookerSeo
```
BookerSeo is not needed anymore as we handle the metadata using app router metadata API (by exporting generateMetadata funtion)

hbjORbj commented on apps/web/modules/users/views/users-type-public-view.tsx:
```diff
@@ -23,40 +22,11 @@ export const getMultipleDurationValue = (
-}: PageProps) {
+function Type({ slug, user, isEmbed, booking, isBrandingHidden, eventData, orgBannerUrl }: PageProps) {
   const searchParams = useSearchParams();
-  const { profile, users, hidden, title } = eventData;
```
no longer used

hbjORbj commented on apps/web/playwright/dynamic-booking-pages.e2e.ts:
```diff
@@ -11,6 +11,8 @@ import {
   selectSecondAvailableTimeSlotNextMonth,
 } from "./lib/testUtils";
 
+test.describe.configure({ mode: "parallel" });
```
small optimization

hbjORbj commented on apps/web/playwright/organization/organization-settings.e2e.ts:
```diff
@@ -66,7 +66,7 @@ async function verifyRobotsMetaTag({ page, orgSlug, urls, expectedContent }: Ver
     for (const relativeUrl of urls) {
       const { url } = await goToUrlWithErrorHandling(relativeUrl);
-      const metaTag = await page.locator('head > meta[name="robots"]');
+      const metaTag = await page.locator('head > meta[name="robots"]').first();
```
Because we didn't completely deprecate `NextSeo` yet (we can completely deprecate once we are 100% done with migration), we can have two metadata tags with `robots` in the html.

hbjORbj commented on apps/web/server/lib/[user]/getServerSideProps.ts:
```diff
@@ -132,6 +133,7 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
   const [user] = usersInOrgContext; //to be used when dealing with single user, not dynamic group
 
   const profile = {
+    id: user.id,
```
passing id here to get user's avatar url (needed for OG image generation in generateMetadata function)

hariombalhara replied:
Can we not use the image at line 138 directly?

hbjORbj replied:
oh nice, good catch :) addressed that

hbjORbj committed:
Merge branch 'main' into chore/app-router-team-pages

hbjORbj committed:
move embed pages back to /future

hbjORbj committed:
add back embed pages in pages router

hbjORbj committed:
revert some changes

hbjORbj commented:
Some notes to make your review easier: @zomars @emrysal 

- Embed pages will still be rendered by Pages Router because the embed pages in App Router are still inside `/future` directory.
- Since metadata is handled by App Router's built-in Metadata API (generateMetadata function), I removed `HeadSeo` from booking components as well as `BookerSeo`. I think this should be fine because embed pages should not need metadata.
- d/[link]/[slug] is already migrated to App Router in Production. In this PR, I removed `HeadSeo` from its client component as this is now handled by App router's Metadata API

hbjORbj commented on apps/web/app/layout.tsx:
```diff
@@ -26,7 +26,10 @@ export const generateMetadata = () => prepareRootMetadata();
   const { pathname, searchParams } = new URL(url);
 
-  const isEmbed = pathname.endsWith("/embed") || (searchParams?.get("embedType") ?? null) !== null;
+  const isEmbedSnippetGeneratorPath = pathname.startsWith("/event-types");
+  const isEmbed =
+    (pathname.endsWith("/embed") || (searchParams?.get("embedType") ?? null) !== null) &&
+    !isEmbedSnippetGeneratorPath;
   const embedColorScheme = searchParams?.get("ui.color-scheme");
```
I synced this file with `_document.tsx` in pages router

hbjORbj committed:
fix import type checks

hbjORbj committed:
Merge branch 'main' into chore/app-router-team-pages

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into chore/app-router-team-pages

hbjORbj committed:
simplify

hbjORbj committed:
fix

hariombalhara committed:
feat: Implement generateBookingPageMetadata function for improved SEO and metadata handling across user and event pages (#18440)

- Added a new utility function `generateBookingPageMetadata` to streamline the generation of metadata for booking and user profile pages.
- Updated multiple page components to utilize the new function, enhancing SEO indexing and metadata consistency.
- Removed redundant code related to previous metadata generation methods, improving code clarity and maintainability.

hbjORbj committed:
fix dirs

hariombalhara committed:
Pr-review-fixes-app-router-team-pages (#18450)

* Remove unnecessary getPublicEvent call

* Remove unused variable

hariombalhara committed:
Fix 404 for team page (#18451)

hariombalhara reviewed (dismissed):
LGTM !! except one thing cc @Udit-takkar 

Could you verify if instant-meeting endpoint related changes are correct?

http://acme.cal.local:3000/instant-meeting/team/team1/team1-event-1 - This doesn't work(404)
But this works
http://acme.cal.local:3000/org/acme/instant-meeting/team/team1/team1-event-1

hariombalhara committed:
Remove console log

Udit-takkar commented:
> Could you verify if instant-meeting endpoint related changes are correct?

Yep. All working fine
